javers-core/src/test/java/org/javers/core/examples/ComparingTopLevelSetVoExample.java

```diff
@@ -0,0 +1,98 @@
+package org.javers.core.examples;
+
+import java.util.Set;
+
+import org.javers.core.Javers;
+import org.javers.core.JaversBuilder;
+import org.javers.core.diff.Diff;
+import org.javers.core.diff.changetype.PropertyChangeType;
+import org.javers.core.diff.changetype.ValueChange;
+import org.javers.core.diff.changetype.container.SetChange;
+import org.javers.core.examples.vo.Person;
+import org.javers.core.graph.HashCodeObjectHasher;
+import org.junit.Test;
+
+import static org.fest.assertions.api.Assertions.assertThat;
+
+/**
+ * @author tomas.kupka
+ */
+public class ComparingTopLevelSetVoExample {
+
+    private Set<Person> oldSet = Set.of(new Person("tommy", "Tommy Smart"), new Person("tommy2", "Tommy Smart2"));
+    private Set<Person> newSet = Set.of(new Person("tommy2", "Tommy Smart2"), new Person("tommy", "Tommy C. Smart"));
+
+    @Test
+    public void shouldDeeplyCompareTwoTopLevelCollectionsWithDefaultObjectHasher() {
+        //given
+        Javers javers = JaversBuilder.javers()
+            .build();
+
+        //when
+        Diff diff = javers.compareCollections(oldSet, newSet, Person.class);
+
+        System.out.println(diff.prettyPrint());
+
+        assertThat(diff.getChanges()).hasSize(5);
+
+        //then
+        //there should be one change of type {@link ValueChange}
```

Comment:
there should be one change ?

Reply:
Definitely not. It should be 4. Just remained from the original test down below